### ISSUE TITLE
Date format abbreviated month not used in es_US locale

### ISSUE THREAD

**Issue by rodcheater** [flutter/flutter#117438]
The abbreviated month is not being used in `es_US` locale when using the skeleton pattern.

Try this code:

```
import 'package:intl/date_symbol_data_local.dart';
import 'package:intl/intl.dart';

void main() {
  initializeDateFormatting();
  print(DateFormat('d MMM yyyy', 'es_US').format(DateTime.now()));
  print(DateFormat.yMMMd('es_US').format(DateTime.now()));
}
```

It prints:

```
21 dic. 2022
21 de diciembre de 2022
```

Note that instead using `es_ES` as the locale produces the expected result.

**Comment by rodcheater** [comment]
Never mind, this is a Dart issue. 🤦

https://github.com/dart-lang/sdk/issues/50811

**Comment by rodcheater** [comment]
Ha ha ha ha.

Turns out this is also a Flutter issue.

Try this code with `intl` at version 0.18.0:

```
import 'dart:ui';

import 'package:flutter_localizations/flutter_localizations.dart';
import 'package:intl/date_symbol_data_local.dart';
import 'package:intl/intl.dart';

void main() {
  initializeDateFormatting();
  print(DateFormat('d MMM yyyy', 'es_US').format(DateTime.now()));
  print(DateFormat.yMMMd('es_US').format(DateTime.now()));

  GlobalCupertinoLocalizations.delegate.load(const Locale('es', 'US'));
  print(DateFormat('d MMM yyyy', 'es_US').format(DateTime.now()));
  print(DateFormat.yMMMd('es_US').format(DateTime.now()));
}
```

It prints:

```
30 dic 2022
30 dic 2022
30 dic 2022
30 de diciembre de 2022
```

Apparently Flutter has it own pattern codes, which I guess need to be updated to match `intl`.

**Comment by danagbemava-nc** [comment]
Hi @rodcheater, I'm curious, what version of `intl` is in your `pubspec.lock`? `flutter_localizations` currently has `intl` pinned to version `0.17.0` (see   https://github.com/flutter/flutter/blob/3223972956595acc37727ed599ba5fa75d96b6a7/packages/flutter_localizations/pubspec.yaml#L11), so I think those 2 versions wouldn't be compatible in your project (at least I couldn't resolve dependencies when I tried to add intl : ^1.18.0 and flutter_localizations)

<details>
<summary>logs</summary>

```
Running "flutter pub get" in ddt...
Resolving dependencies...
Because ddt depends on flutter_localizations from sdk which depends on intl 0.17.0, intl 0.17.0 is required.
So, because ddt depends on intl 0.18.0, version solving failed.
pub get failed
command: "/Users/nexus/dev/sdks/flutters/bin/cache/dart-sdk/bin/dart __deprecated_pub --directory . get --example"
pub env: {
  "FLUTTER_ROOT": "/Users/nexus/dev/sdks/flutters",
  "PUB_ENVIRONMENT": "vscode.dart-code:flutter_cli:get",
  "PUB_CACHE": "/Users/nexus/.pub-cache",
}
exit code: 1
```
</details>

**Comment by rodcheater** [comment]
Yes, I have an override to force it to 0.18.0. The bad date format pattern is fixed in `intl` 0.18.0, but `flutter_localisations` has its [own version](https://github.com/flutter/flutter/blob/3223972956595acc37727ed599ba5fa75d96b6a7/packages/flutter_localizations/lib/src/l10n/generated_date_localizations.dart#L18989) so, despite the override, I still see the bad date format.

**Comment by danagbemava-nc** [comment]
Thanks for the clarification. 

From the intl package: https://github.com/dart-lang/intl/blob/master/lib/date_time_patterns.dart#L1627
the format is    `'yMMMd': 'd MMM y',`

Comparing with the current version in flutter_localizations: https://github.com/flutter/flutter/blob/master/packages/flutter_localizations/lib/src/l10n/generated_date_localizations.dart#L19011

we have  `'yMMMd': "d 'de' MMMM 'de' y",`

Labeling for further investigation from the team.

**Comment by goderbauer** [comment]
cc @thkim1011
